### PR TITLE
Ruby

### LINKED ISSUE #408: New Practice: Gitignore practice for Ruby
## New Practice Proposal

Create a new practice that checks correctly set `.gitignore` for Ruby.

### PR DESCRIPTION
**Pull request by king-11**
## Description

Gitignore correctly set for ruby only necessary components included for check

## Motivation and Context

closes #408 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have updated the documentation accordingly.
- [ ] I have added new practice to practice list in README.md.
- [x] I have read the **CONTRIBUTING** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/practices/Ruby/RubyGitignoreCorrectlySetPractice.spec.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+const basicGitignore = `
+*.gem
+*.rbc
+/coverage/
+/tmp/
+/.config
+`;
+
+describe('CSharpGitignoreCorrectlySetPractice', () => {
```
Comment (adelkahomolova) [inline src/practices/Ruby/RubyGitignoreCorrectlySetPractice.spec.ts:14]:
```suggestion
describe('RubyGitignoreCorrectlySetPractice', () => {
```

File: src/practices/Ruby/RubyGitignoreCorrectlySetPractice.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+@DxPractice({
+  id: 'Ruby.GitignoreCorrectlySet',
+  name: 'Set .gitignore Correctly',
+  impact: PracticeImpact.high,
+  suggestion: 'Set patterns in the .gitignore as usual',
+  reportOnlyOnce: true,
+  url: 'https://github.com/github/gitignore/blob/master/Ruby.gitignore',
+  dependsOn: { practicing: ['LanguageIndependent.GitignoreIsPresent'] },
+})
+export class CSharpGitignoreCorrectlySetPractice extends PracticeBase {
```
Comment (adelkahomolova) [inline src/practices/Ruby/RubyGitignoreCorrectlySetPractice.ts:17]:
```suggestion
export class RubyGitignoreCorrectlySetPractice extends PracticeBase {
```

**Review by adelkahomolova — CHANGES_REQUESTED** [review]
Change the names from CSharp to Ruby, please. It's CSharp everywhere :(

**Comment by adelkahomolova** [comment]
And please - change also the name of Pull Request. @king-11

**Comment by king-11** [comment]
@adelkahomolova am really sorry I was just in sync to get this done so missed upon names :see_no_evil:

**Review by adelkahomolova — APPROVED** [review]

**Review by prokopsimek — APPROVED** [review]
@king-11 Thank you! It's an awesome pull request. 👏🤠

**Pull request merged**